### PR TITLE
feat: in W3C header injection: add option to request even if does'nt exists

### PR DESCRIPTION
**Pull request by saartochner**

### CODE REVIEW

File: src/hooks/http.ts [hunk 1]
```diff
@@ -123,6 +123,29 @@ export class Http {
...
 
+  static addOptionsToHttpRequestArguments(originalArgs, newOptions) {
+    // We're switching on the different signatures of http:
+    // https://nodejs.org/api/http.html#httpgeturl-options-callback
+    if (typeof originalArgs[0] === 'string' || originalArgs[0] instanceof URL) {
+      if (originalArgs[1]) {
+        if (typeof originalArgs[1] === 'function') {
+          // The signature is: (url, callback)
+          originalArgs.push(originalArgs[1]);
+          originalArgs[1] = newOptions;
```
Comment (nirLumigo) [inline src/hooks/http.ts:134]: For each option, can you also add a comment for the new signature?

File: src/hooks/http.ts [hunk 2]
```diff
@@ -191,11 +214,14 @@ export class Http {
       if (isRequestToAwsService && !isKeepHeadersOn()) {
         const { awsXAmznTraceId } = getAWSEnvironment();
         const traceId = getPatchedTraceId(awsXAmznTraceId);
-        options.headers['X-Amzn-Trace-Id'] = traceId;
+        headers && (headers['X-Amzn-Trace-Id'] = traceId);
```
Comment (nirLumigo) [inline src/hooks/http.ts:217]: Was this change needed? there is a default value for headers in line 205

File: src/hooks/http.test.js [hunk 3]
```diff
@@ -838,6 +875,49 @@ describe('http hook', () => {
     expect(span.info.httpInfo.request.headers[TRACESTATE_HEADER_NAME]).toBeDefined();
   });
 
+  test('wrapHttpLib - check request - headers exists', () => {
```
Comment (nirLumigo) [inline src/hooks/http.test.js:878]: What is the purpose of the test? Verifying that `TRACESTATE_HEADER_NAME` exists? Please add it in the description of the test. Same regarding the next test.

**Review by nirLumigo — APPROVED** [review]
Nice! :)
Added small comments, please check before merging.

File: src/hooks/http.ts (hunk 2 above)
Comment (saartochner) [reply src/hooks/http.ts:217]: Not use options is the important change here :)
